sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/CallMethodViaReflection.scala

```diff
@@ -55,7 +55,7 @@ import org.apache.spark.util.Utils
 case class CallMethodViaReflection(children: Seq[Expression])
   extends Expression with CodegenFallback {
 
-  override def prettyName: String = "reflect"
```

Comment:
Shall we just override the node name?

Reply:
Actually I was thinking about this one. Seems few expressions override `nodeName` directly but I am not sure if that's correct. Node name seems to be usually the class name; not lower-cased names. Fixing `prettyName` looks good enough to address the issue in output column names.

Reply:
It's good enough, but I don't think `nodeName` is designed to represent the class name. For example, `TreeNode.asCode` expects `nodeName` to be the function name.

Reply:
But `asCode` says:

> Returns a 'scala code' representation of this `TreeNode` and its children.

Reply:
damn I miss read it. Then it won't work for physical plans as we remove the `Exec` postfix.

Reply:
anyway my point is: `nodeName` is used in multiple string methods `prettyName`, `simpleString`, etc. Seems better to make them consistent.

Reply:
Yes, I agree with that point. But here why don't we just don't touch `nodeName` and revisit next time?

If we override `prettyName`, most of cases such as `simpleString` and `sql` and will be addressed.

My impression about `prettyName` via `sql` call is user-facing string (as it's used in `sql` as of https://github.com/apache/spark/commit/d9efe63ecdc60a9955f1924de0e8a00bcb6a559d). and `nodeName` seems more like related to internal and debugging. If we have to debug, we should take a look for class names rather then what alias was used by the user.

Reply:
ok makes sense